pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -25,6 +32,155 @@ import (
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		objects := []runtime.Object{
+			&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "serving-cert", Namespace: "openshift-controller-manager"}},
+			&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "etcd-client", Namespace: "kube-system"}},
+			&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "client-ca", Namespace: "openshift-kube-apiserver"}},
+		}
+		kubeClient := fake.NewSimpleClientset(objects...)
+		cv := &configv1.ClusterVersion{}
+
+		indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+		proxyLister := configlistersv1.NewProxyLister(indexer)
+
+		operatorConfig := &operatorv1.OpenShiftControllerManager{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "cluster",
+				Generation: cv.Generation,
+			},
+			Spec: operatorv1.OpenShiftControllerManagerSpec{
+				OperatorSpec: operatorv1.OperatorSpec{},
+			},
+			Status: operatorv1.OpenShiftControllerManagerStatus{
+				OperatorStatus: operatorv1.OperatorStatus{},
+			},
+		}
+		kubeInformers := v1helpers.NewKubeInformersForNamespaces(kubeClient, "", util.TargetNamespace)
+		configMapsGetter := v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformers)
+		controllerManagerOperatorClient := operatorfake.NewSimpleClientset(operatorConfig)
+
+		indexer.Add(cv)
+		operator := OpenShiftControllerManagerOperator{
+			kubeClient:           kubeClient,
+			configMapsGetter:     kubeClient.CoreV1(),
+			proxyLister:          proxyLister,
+			recorder:             events.NewInMemoryRecorder(""),
+			operatorConfigClient: controllerManagerOperatorClient.OperatorV1(),
+			clusterVersionLister: configlistersv1.NewClusterVersionLister(indexer),
+		}
+		resultMap, _, err := manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(tc.versionLister, tc.versionName, kubeClient, configMapsGetter, operator.recorder, operatorConfig)
+		if err != nil {
+			t.Error(err)
+		}
+		diff := cmp.Diff(tc.result.Data["config.yaml"], resultMap.Data["config.yaml"])
+		if err == nil && tc.result.Data["config.yaml"] != resultMap.Data["config.yaml"] {
```

Comment:
Comparing strings/[]bytes will make reading the diff harder. Also, as you mentioned there's a lot of empty/nil fields. Can you instead create `expectedOpenShiftControllerManagerConfig`, decode `resultMap.Data["config.yaml"]` and cmp.Diff both structs? E.g. [resultMap.Data["config.yaml"] a](https://github.com/openshift/cluster-openshift-apiserver-operator/pull/532/files#diff-25e7c9a2521d15c93e122ab635c1ec19358116650aa275abca5e014b514c69c2R373-R388) and https://github.com/openshift/cluster-openshift-apiserver-operator/pull/532/files#diff-25e7c9a2521d15c93e122ab635c1ec19358116650aa275abca5e014b514c69c2R407-R418

Reply:
I did it a bit differently, please check